ql/src/test/results/clientpositive/llap/explainuser_1.q.out

```diff
@@ -4990,38 +4990,44 @@ Vertex dependency in root stage
 Reducer 2 <- Map 1 (SIMPLE_EDGE)
 Reducer 3 <- Reducer 2 (SIMPLE_EDGE)
 Reducer 4 <- Reducer 3 (SIMPLE_EDGE)
+Reducer 5 <- Reducer 4 (SIMPLE_EDGE)
```

Comment:
I have to say that I was not expecting these changes - in this case the "grouping of key2/key3" is removed; so I'm not sure in the absencce of those columns from partitioning the GBY will work as expected....

I've attached a screenshot from the diff where the changes could be seen better:

![a2](https://user-images.githubusercontent.com/1902540/67288932-109aac80-f4de-11e9-89d8-49d447d833e2.png)

Reply:
Since the plan is a user explain, it does not show the sorting keys in the shuffle. However, group by operator produces correct results if we partition data by first column and sort by the three grouping columns. I believe that is what was happening here. The additional shuffle phase is unnecessary.

Reply:
I agree; that was something I was not considering....I'll change the groupbyoperator acceptor - to recognize gby operators in "complete mode" - right now I think only those which can be accepted